### ISSUE TITLE
Feature Request: Easy way to predict after training model with Estimator and Dataset API

### ISSUE THREAD

**Issue by secsilm** [tensorflow/tensorflow#15327]
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:  Windows 10
- **TensorFlow installed from (source or binary)**: pip
- **TensorFlow version (use command below)**: 1.4.0
- **Python version**: Python 3.5.2 :: Anaconda custom (64-bit)
- **Bazel version (if compiling from source)**: None
- **GCC/Compiler version (if compiling from source)**: None

### Describe the problem

I have beed trained a image classification cnn model with the Estimator and Dataset(`tf.data.TFRecordsDataset`) API. The relative model files have bee saved in `model_dir`. The last few days I try very hard to figure out how to predict one or more images label using the saved model files. 

However I failed and don't know what to do. I can't find relative contents in the official doc. So adding an easy method to do this may be a good idea. Or is there another solution I missed?

FYI, my training code is [here](https://github.com/secsilm/understaing-datasets-estimators-tfrecords/blob/master/cifar10-estimator-dataset.py).

**Comment by sleighsoft** [comment]
Estimator has a `predict` function that accepts an `input_fn`which iteratively produces data to be predicted. This can be data from `Dataset` API or simply numpy arrays. See here https://www.tensorflow.org/versions/master/api_docs/python/tf/estimator/Estimator#predict

**Comment by secsilm** [comment]
@sleighsoft thank you for your reply. Yes I know that Estimators have a `predict` method. But what I want to do is using a separate script to predict when finishing training progress.

**Comment by sleighsoft** [comment]
Maybe this helps you https://www.tensorflow.org/programmers_guide/saved_model#using_savedmodel_with_estimators

**Comment by secsilm** [comment]
It seems that it's TensorFlow Serving staff? I just to want to predict with the trained model, no server staff. I didn't expect it so hard.

**Comment by angersson** [comment]
You may also find [the Abalone tutorial](https://www.tensorflow.org/extend/estimators#running_the_abalone_model) useful, which describes how to run a prediction.

Since your use case is covered, I'm closing this issue (thanks, @sleighsoft, for the help!). If you have more questions about how to use Tensorflow, please check out or ask a question on [Stack Overflow](http://stackoverflow.com/questions/tagged/tensorflow). Thanks!

**Comment by ricoms** [comment]
something like tensorflow_estimator_to_keras, and then use keras easy predict.

**Comment by dksb** [comment]
Are you satisfied with the resolution of your issue?
[Yes](https://goo.gl/forms/Oe0tEvODFRoI2gJF3)
[No](https://goo.gl/forms/fUjzOfrtkFbrOT8d2)

**Comment by ricoms** [comment]
where is that form from @dksb ? will it be considered? by who?
